src/Artifacts.jl

```diff
@@ -786,7 +786,9 @@ function download_artifact(
             msg *= "  Expected git-tree-sha1:   $(bytes2hex(tree_hash.bytes))\n"
             msg *= "  Calculated git-tree-sha1: $(bytes2hex(calc_hash.bytes))"
             @error(msg)
-            return false
+            # Tree hash calculation is still broken on some systems, e.g. Pkg.jl#1860,
+            # so we return true here and only raise the warning on the lines above.
+            # return false
```

Comment:
Don't we need to actually have the return?
```suggestion
            # return false
            return true
```

Reply:
That would result in an early return in case where more code is added later in this function, we just want to keep going, not explicitly return here.